getinsomnia/insomnia
# feat: enable manipulation on insomnia.request - INS-3379

This change basically sets up `insomnia.request` for manipuating request before sending.
- [x] enable `insomnia.request`
- [x] added a test for it 
- [x] add `interfaces.ts` for common interfaces
- [x] aggregate some requests to `pre-request-collection.yaml`

### How to test
Run the following script and check if it works well:
```
                const { Header } = require('insomnia-collection');
                insomnia.request.method = 'GET';
                insomnia.request.url.addQueryParams('k1=v1');
                insomnia.request.headers.add(new Header({
                    key: 'Content-Type',
                    value: 'text/plain'
                }));
                insomnia.request.headers.add(new Header({
                    key: 'X-Hello',
                    value: 'hello'
                }));
                insomnia.request.body.update({
                    mode: 'raw',
                    raw: 'rawContent',
                });
                insomnia.request.auth.update(
                    {
                        type: 'bearer',
                        bearer: [
                                {key: 'token', value: 'tokenValue'},
                        ],
                    },
                    'bearer'
                );
```
